var/spack/repos/builtin/packages/intel-tbb/package.py

```diff
@@ -34,33 +34,45 @@ class IntelTbb(Package):
     """
     homepage = "http://www.threadingbuildingblocks.org/"
 
-    # Only version-specific URL's work for TBB
-    version('2018.2', '0b8dfe30917a54e40828eeb0ed7562ae',
-            url='https://github.com/01org/tbb/archive/2018_U2.tar.gz')
-    version('2018.1', 'b2f2fa09adf44a22f4024049907f774b',
-            url='https://github.com/01org/tbb/archive/2018_U1.tar.gz')
-    version('2018.0', 'e54de69981905ad69eb9cf0226b9bf5f9a4ba065',
-            url='https://github.com/01org/tbb/archive/2018.tar.gz')
-    version('2017.8', '488f049fd107d8b1f6ba59cf4aad881172525106',
-            url='https://github.com/01org/tbb/archive/2017_U8.tar.gz')
-    version('2017.6', 'c0a722fd1ae66b40aeab25da6049086ef5f02f17',
-            url='https://github.com/01org/tbb/archive/2017_U6.tar.gz')
-    version('2017.5', '26f720729d322913912e99d1e4a36bd10625d3ca',
-            url='https://github.com/01org/tbb/archive/2017_U5.tar.gz')
-    version('2017.3', '2c451a5bcf6fc31487b98b4b29651c369874277c',
-            url='https://www.threadingbuildingblocks.org/sites/default/files/software_releases/source/tbb2017_20161128oss_src.tgz')
-    version('4.4.4', 'd4cee5e4ca75cab5181834877738619c56afeb71',
-            url='https://www.threadingbuildingblocks.org/sites/default/files/software_releases/source/tbb44_20160413oss_src.tgz')
-    version('4.4.3', '80707e277f69d9b20eeebdd7a5f5331137868ce1',
-            url='https://www.threadingbuildingblocks.org/sites/default/files/software_releases/source/tbb44_20160128oss_src_0.tgz')
+    # See url_for_version() below.
+    version('2018.3', 'cd2e136598ffa5c136f077ee85a35b4c')
+    version('2018.2', '0b8dfe30917a54e40828eeb0ed7562ae')
+    version('2018.1', 'b2f2fa09adf44a22f4024049907f774b')
+    version('2018',   '7fb30d5ea2545f26ce02757d9ab05e6c')
+    version('2017.8', '7240f57f1aeea0e266a5e17ae68fdc16')
+    version('2017.7', '364f2a4b80e978f38a69cbf7c466b898')
+    version('2017.6', 'ec21254af4fc2a29574c272f501a3138')
+    version('2017.5', '85b41c64102c052e24d8a39f6193e599')
+    version('2017.4', '71526b2fef098515e212302d1455de7d')
+    version('2017.3', 'd7622eeaafeff8d271c7aa684bd82ddb')
+    version('2017.2', '9605cbea96998a10a186fc72c35cbd76')
+    version('2017.1', '6c0fe8aa7bc911a85e8e522e620511b3')
+    version('2017',   '9e7f9ea684ecf84ac74dcd3c6012cfa6')
+    version('4.4.6',  '20e15206f70c2651bfc964e451a443a0')
+    version('4.4.5',  '531a67cd98f9b4ec8ece95c5f8193a83')
+    version('4.4.4',  '61531b2e8684e06a621dcdca1a7a420e')
+    version('4.4.3',  '8e3e39e1fdfb3f7c3a5ac8ec1afe186e')
+    version('4.4.2',  'e92b110e8eb238741b00e3789b39969e')
+    version('4.4.1',  'a02c9958f02c1b5f3626874219979ae8')
+    version('4.4',    '1d512085221996eae6cec04e1a4cd3dd')
 
     provides('tbb')
 
+    # Assembler version is important.
+    depends_on('binutils', type='build', when='platform=linux %gcc@4.8.0:')
```

Comment:
See http://spack.readthedocs.io/en/latest/getting_started.html?highlight=binutils#binutils

Reply:
Unsure quite what to do here based on advice at link. The conditionally-applied patch mitigates the problem with older `binutils`, but I either apply it unconditionally (previous behavior) with efficiency implications, or have a version of `binutils` against which to compare to ascertain whether the patch should be applied or not. I could just remove the patch entirely, but then how does one let the user know what to do in case of a build failure (e.g. on Centos6-based systems)? Please advise.